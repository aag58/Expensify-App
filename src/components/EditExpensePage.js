import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import { editExpenses, startRemoveExpenses} from '../actions/expenses';


// export class EditExpensePage extends React.Component{
//     onSubmit= (expense)=>{
//         this.props.editExpenses(this.props.expense.id, expense);
//         this.props.history.push('/');
//     }
//     onRemove = ()=>{
//         this.props.removeExpense(this.props.expense);
//         this.props.history.push('/'); 
//     }

//     render(){
//         return(
//             <div>
//                 <ExpenseForm 
//                     expense = {this.props.expense}
//                     onSubmit={this.onSubmit}
//                 />
//                 <button onClick={this.onRemove}>Remove</button>
//             </div>
//         )
//     }
// }

export const EditExpensePage = (props)=>{
    return(
        <div>
            <ExpenseForm 
                expense = {props.expense}
                onSubmit={(expense)=>{
                    props.editExpenses(props.expense.id, expense);
                    props.history.push('/');
                }}
            />
            <button onClick={()=>{
                props.startRemoveExpenses(props.expense)
                props.history.push('/');
                
                
            }}>Remove</button>

        </div>
    )
}

const mapStateToProps = (state, props)=>({
    expense: state.expenses.find((expense)=> expense.id===props.match.params.id) 
})

// const mapDispatchToProps = (dispatch, props)=>({
//     editExpenses: (id, expense) => dispatch(editExpenses(id, expense)),
//     startRemoveExpenses: (data) => dispatch(startRemoveExpenses(data))
// })
export default connect(mapStateToProps, {editExpenses, startRemoveExpenses})(EditExpensePage);

