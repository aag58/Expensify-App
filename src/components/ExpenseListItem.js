import React from 'react';
import {removeExpense} from '../actions/expenses';
import { NavLink } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
// const ExpenseListItem = (props)=>(
//     <div>
//         <h3>{props.description}</h3>
//         <p>{props.amount}-{props.createdAt}</p>
//         <button onClick={()=>{
//             props.dispatch(removeExpense(props))
//         }}>Remove</button>
        
//     </div>
// )
const ExpenseListItem = ({id,description, amount,createdAt})=>(
    <div>
        <NavLink to={`/edit/${id}`}><h3>{description}</h3></NavLink>
        <p>
        {numeral(amount/100).format('$0,0.00')}
        -
        {moment(createdAt).format('MMMM Do, YYYY')}
        </p>               
    </div>
)


export default ExpenseListItem;