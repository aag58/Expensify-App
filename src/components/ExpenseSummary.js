import React from 'react';
import {connect} from 'react-redux'
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses'; 
import selectExpenseTotal from '../selectors/expense-total';

export const ExpenseSummary = ({expensesCount, expensesTotal, visibleExpenses})=>{
    const expenseWord = expensesCount ===1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numeral(expensesTotal/100).format('$0,0.00')  
    return(
        <div>
            <h1>Viewing {expensesCount} {expenseWord} totaling {formattedExpenseTotal}</h1>
        </div>
        
    )    
}

const mapStateToProps = (state)=>{
    const visibleExpenses= selectExpenses(state.expenses, state.filters)
    return{
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpenseTotal(visibleExpenses)        
    }
}
export default connect(mapStateToProps)(ExpenseSummary)