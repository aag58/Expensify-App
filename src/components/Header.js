import React from 'react';
import {NavLink} from 'react-router-dom';
const Header = ()=>(
    <div>
        <h1>Expensify</h1>
        <p>
            <NavLink exact to="/" activeClassName="is-active">home page</NavLink>
            <NavLink to="/create" activeClassName="is-active">Add Expense Page</NavLink>
            
        </p>
    </div>
    
)
export default Header;