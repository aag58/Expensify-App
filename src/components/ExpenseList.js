import React from 'react';
import {connect}  from 'react-redux';
import ExpenseListItem from './ExpenseListItem';

import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props)=> {
    // const ExpenseListItems = props.expenses.map((expense)=> (<ExpenseListItem 
    //                                                             key={expense.id} 
    //                                                             {...expense}
    //                                                             />)
    //                                             )
    return(
       
        <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                   props.expenses.map((expense)=> (<ExpenseListItem 
                    key={expense.id} 
                    {...expense}
                    />)
                )
            )
        }
            
            
        </div>
    )    
}



const mapStateToProps = (state)=>{
    return{
        expenses: selectExpenses(state.expenses, state.filters)
        // filters: state.filters
    }
    
}

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList)

export default ConnectedExpenseList;