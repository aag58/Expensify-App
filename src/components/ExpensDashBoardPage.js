import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpensDashBoardPage = ()=> (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
)


export default ExpensDashBoardPage;