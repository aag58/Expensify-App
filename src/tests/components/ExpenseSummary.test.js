import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Should render expense summary correctly with one expense', ()=>{
    const wrapper = shallow(<ExpenseSummary expensesCount = {1} expensesTotal = {235}/>)
    expect(wrapper).toMatchSnapshot();

})


test('Should render expense summary correctly with multiple expense', ()=>{
    const wrapper = shallow(<ExpenseSummary expensesCount ={23} expensesTotal={235456534}/>)
    expect(wrapper).toMatchSnapshot();
})