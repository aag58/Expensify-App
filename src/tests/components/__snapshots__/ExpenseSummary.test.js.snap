// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense summary correctly with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totaling 
    $2,354,565.34
  </h1>
</div>
`;

exports[`Should render expense summary correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
